federation/pkg/federation-controller/service/servicecontroller.go

```diff
@@ -0,0 +1,830 @@
+	//return cloudprovider.GetDNSNameFor(service) // TODO: quinton: Should not be DNS provider specific.  Write a generic one compliant with the RFC.
+	return ""
+}
+
+func getPortsForLB(service *v1.Service) ([]*v1.ServicePort, error) {
+	// TODO: quinton: Probably applies for DNS SVC records.  Come back to this.
+	//var protocol api.Protocol
+
+	ports := []*v1.ServicePort{}
+	for i := range service.Spec.Ports {
+		sp := &service.Spec.Ports[i]
+		// The check on protocol was removed here.  The DNS provider itself is now responsible for all protocol validation
+		ports = append(ports, sp)
+		//TODO:jesse
+		//if protocol == "" {
+		//	protocol = sp.Protocol
+		//} else if protocol != sp.Protocol && wantsLoadBalancer(service) {
+		//	// TODO:  Convert error messages to use event recorder
+		//	return nil, fmt.Errorf("mixed protocol external load balancers are not supported.")
+		//}
+	}
+	return ports, nil
+}
+
+func portsEqualForLB(x, y *v1.Service) bool {
+	xPorts, err := getPortsForLB(x)
+	if err != nil {
+		return false
+	}
+	yPorts, err := getPortsForLB(y)
+	if err != nil {
+		return false
+	}
+	return portSlicesEqualForLB(xPorts, yPorts)
+}
+
+func portSlicesEqualForLB(x, y []*v1.ServicePort) bool {
+	if len(x) != len(y) {
+		return false
+	}
+
+	for i := range x {
+		if !portEqualForLB(x[i], y[i]) {
+			return false
+		}
+	}
+	return true
+}
+
+func portEqualForLB(x, y *v1.ServicePort) bool {
+	// TODO: Should we check name?  (In theory, an LB could expose it)
+	if x.Name != y.Name {
+		return false
+	}
+
+	if x.Protocol != y.Protocol {
+		return false
+	}
+
+	if x.Port != y.Port {
+		return false
+	}
+	// we do not need to check node port for federation
+	//if x.NodePort != y.NodePort {
+	//	return false
+	//}
+
+	// We don't check TargetPort; that is not relevant for dns
+	// TODO: Should we blank it out?  Or just check it anyway?
+
+	return true
+}
+
+func clustersFromList(list *v1alpha1.ClusterList) []string {
+	result := []string{}
+	for ix := range list.Items {
+		result = append(result, list.Items[ix].Name)
+	}
+	return result
+}
+
+// getClusterConditionPredicate filter all clusters meet condition of
+// condition.type=Ready and condition.status=true
+func getClusterConditionPredicate() federationcache.ClusterConditionPredicate {
+	return func(cluster v1alpha1.Cluster) bool {
+		// If we have no info, don't accept
+		if len(cluster.Status.Conditions) == 0 {
+			return false
+		}
+		for _, cond := range cluster.Status.Conditions {
+			//We consider the cluster for load balancing only when its ClusterReady condition status
+			//is ConditionTrue
+			if cond.Type == v1alpha1.ClusterReady && cond.Status != v1.ConditionTrue {
+				glog.V(4).Infof("Ignoring cluser %v with %v condition status %v", cluster.Name, cond.Type, cond.Status)
+				return false
+			}
+		}
+		return true
+	}
+}
+
+// clusterServiceSyncLoop handles updating the dns pointed to by all federated service DNS Records
```

Comment:
Name in comment doesn't match name of function.  
Also, rename the function to reflect better what is does, and make the comment consistent with that.